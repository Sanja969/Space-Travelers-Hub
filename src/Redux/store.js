import { configureStore } from '@reduxjs/toolkit';
import { combineReducers } from 'redux';
import thunk from 'redux-thunk';
import logger from 'redux-logger';
import rocketsReducer from './rocket-redux';
import missionsReducer from './missions-redux';

const rootReducer = combineReducers({
  rockets: rocketsReducer,
  missions: missionsReducer,
});

const store = configureStore({
  reducer: rootReducer,
  middleware: [thunk, logger],
});

export default store;

// console.log(store);
