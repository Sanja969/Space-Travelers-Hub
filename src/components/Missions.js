import React from 'react';
import Singlemission from './MissionList';

const Missions = () => (
  <div className="mt-5">
    <Singlemission />
  </div>
);
export default Missions;
