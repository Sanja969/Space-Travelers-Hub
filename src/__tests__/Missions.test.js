import React from 'react';
import { Provider } from 'react-redux';
import { render } from '@testing-library/react';
import store from '../Redux/store';
import Missions from '../components/Missions';

describe('Test Missions render', () => {
  test('Test Missions snapshot', () => {
    const tree = render(
      <Provider store={store}>
        <Missions />
      </Provider>,
    );
    expect(tree).toMatchSnapshot();
  });
});
