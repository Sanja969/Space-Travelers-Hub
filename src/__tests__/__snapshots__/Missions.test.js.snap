// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Missions render Test Missions snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="mt-5"
      >
        <div
          class="d-flex justify-content-center px-3 py-5 mt-5"
        >
          <table
            class="table table-bordered table-striped p-0 m-0"
          >
            <thead
              class="thred"
            >
              <tr>
                <th
                  scope="col"
                >
                  Mission
                </th>
                <th
                  scope="col"
                >
                  Description
                </th>
                <th
                  scope="col"
                >
                  Status
                </th>
                <th
                  scope="col"
                />
              </tr>
            </thead>
          </table>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="mt-5"
    >
      <div
        class="d-flex justify-content-center px-3 py-5 mt-5"
      >
        <table
          class="table table-bordered table-striped p-0 m-0"
        >
          <thead
            class="thred"
          >
            <tr>
              <th
                scope="col"
              >
                Mission
              </th>
              <th
                scope="col"
              >
                Description
              </th>
              <th
                scope="col"
              >
                Status
              </th>
              <th
                scope="col"
              />
            </tr>
          </thead>
        </table>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
